app/pages/home-for-user/index.jsx

```diff
@@ -122,40 +122,42 @@ export default class HomePageForUser extends React.Component {
-        activePreferences = activePreferences.map((preference, i) => {
-          preference.sort_order = i;
-          return preference;
-        });
-        this.getProjectsForPreferences(activePreferences)
-          .then((projects) => {
-            return projects
-            .sort((a, b) => {
-              return a.sort_order - b.sort_order;
+        if (activePreferences.length > 0) {
+          activePreferences = activePreferences.map((preference, i) => {
+            preference.sort_order = i;
+            return preference;
+          });
+          this.getProjectsForPreferences(activePreferences)
+            .then((projects) => {
+              return projects
+              .sort((a, b) => {
+                return a.sort_order - b.sort_order;
+              })
+              .filter(Boolean)
+              .map((project, i) => {
+                return {
+                  avatar_src: projects[i].avatar_src,
+                  id: projects[i].id,
+                  slug: projects[i].slug,
+                  display_name: projects[i].display_name,
+                  description: projects[i].description,
+                  color: getColorFromString(projects[i].slug),
+                  classifications: projects[i].activity_count,
+                  updated_at: projects[i].updated_at,
+                  redirect: projects[i].redirect
+                };
+              });
             })
-            .filter(Boolean)
-            .map((project, i) => {
-              return {
-                avatar_src: projects[i].avatar_src,
-                id: projects[i].id,
-                slug: projects[i].slug,
-                display_name: projects[i].display_name,
-                description: projects[i].description,
-                color: getColorFromString(projects[i].slug),
-                classifications: projects[i].activity_count,
-                updated_at: projects[i].updated_at,
-                redirect: projects[i].redirect
-              };
-            });
-          })
-          .then((projects) => {
-            this.setState((prevState) => {
-              const ribbonData = prevState.ribbonData.concat(projects);
-              const totalClassifications = ribbonData.reduce((total, project) => {
-                return total + project.classifications;
-              }, 0);
-              return { ribbonData, totalClassifications };
+            .then((projects) => {
+              this.setState((prevState) => {
+                const ribbonData = prevState.ribbonData.concat(projects);
+                const totalClassifications = ribbonData.reduce((total, project) => {
+                  return total + project.classifications;
+                }, 0);
+                return { ribbonData, totalClassifications };
+              });
             });
-          });
+        }
         const meta = projectPreferences[0].getMeta();
```

Comment:
```suggestion
            const meta = projectPreferences[0].getMeta();
        }
```

Reply:
I think this changes the scope of `meta` so that it's undefined outside the if block. I'll have a look.

Reply:
I've edited the code for clarity, and added some comments. Changing the scope of `meta` would break the paged requests for UPP, but I've moved that code to the beginning of the if block, to separate it from the requests for projects.